lib/logflare/sources.ex

```diff
@@ -40,31 +41,50 @@ defmodule Logflare.Sources do
 
   @spec create_source(map(), User.t()) :: {:ok, Source.t()} | {:error, Ecto.Changeset.t()}
   def create_source(source_params, user) do
+    postgres_backend_adapter_url =
+      Application.get_env(:logflare, :postgres_backend_adapter, []) |> Keyword.get(:url)
+
+    source_params =
+      source_params
+      |> Enum.map(fn {k, v} -> {to_string(k), v} end)
+      |> Map.new()
+      |> then(&Map.put(&1, "v2_pipeline", !!postgres_backend_adapter_url))
+
     source =
       user
       |> Ecto.build_assoc(:sources)
       |> Source.update_by_user_changeset(source_params)
       |> Repo.insert()
 
-    case source do
-      {:ok, source} ->
-        init_schema = SchemaBuilder.initial_table_schema()
+    case postgres_backend_adapter_url do
+      nil -> create_big_query_schema_and_start_source(source)
+      url -> create_backend_adaptor(source, url)
+    end
+  end
 
-        {:ok, _source_schema} =
-          SourceSchemas.create_source_schema(source, %{
-            bigquery_schema: init_schema,
-            schema_flat_map: SchemaUtils.bq_schema_to_flat_typemap(init_schema)
-          })
+  defp create_big_query_schema_and_start_source({:ok, source}) do
+    init_schema = SchemaBuilder.initial_table_schema()
 
-        Source.Supervisor.start_source(source.token)
+    {:ok, _source_schema} =
+      SourceSchemas.create_source_schema(source, %{
+        bigquery_schema: init_schema,
+        schema_flat_map: SchemaUtils.bq_schema_to_flat_typemap(init_schema)
+      })
 
-        {:ok, source}
+    Source.Supervisor.start_source(source.token)
 
-      {:error, changeset} ->
-        {:error, changeset}
-    end
+    {:ok, source}
   end
 
+  defp create_big_query_schema_and_start_source({:error, changeset}), do: {:error, changeset}
+
+  defp create_backend_adaptor({:ok, source}, url) do
+    {:ok, _source_backend} = Backends.create_source_backend(source, :postgres, %{url: url})
```

Comment:
should we compile the repo module here as well and create the conn?

Reply:
unsure honestly, at the moment we're only doing all of that on first ingestion we can change this at a later date